src/Microsoft.PowerShell.Commands.Management/commands/management/Service.cs

```diff
@@ -2015,10 +2015,11 @@ protected override void BeginProcessing()
                         dwStartType = NativeMethods.SERVICE_DISABLED;
                         break;
                     default:
-                        Diagnostics.Assert(
-                            ((ServiceStartMode)(-1)) == StartupType,
-                            "bad StartupType");
-                        break;
+                        WriteNonTerminatingError(StartupType.ToString(), DisplayName, Name,
+                            new ArgumentException(), "CouldNotNewService",
+                            ServiceResources.UnsupportedStartupType,
```

Comment:
There is another instance of `Assert(((ServiceStartMode)(-1)) == StartupType, "bad StartupType"` in `Set-Service`. It would be the best if all can be fixed here.

Reply:
will fix